cucumber/rotators/features/password.feature

```diff
@@ -0,0 +1,31 @@
+Feature: Postgres password rotation
+
+  Background: Configure a postgres rotator
+    Given I reset my root policy
+    And I have the root policy:
+    """
+    - !policy
+       id: db-reports
+       body:
+         - &variables
+           - !variable url
+           - !variable username
+           - !variable
+             id: password
+             annotations:
+               rotation/rotator: postgresql/password
+               rotation/ttl: PT1S
+               rotation/postgresql/password/length: 32
+    """
+    And I create a db user "test" with password "secret"
+    And I watch for changes in "db-reports/password" and db user "test"
+    And I add the value "testdb" to variable "db-reports/url"
+    And I add the value "test" to variable "db-reports/username"
+    And I add the value "secret" to variable "db-reports/password"
+
+  Scenario: Values are rotated according to the policy
+    Given I wait for 3 seconds
```

Comment:
Like we talked about before, I think you're asking for trouble by making your tests depend on something happening within a particular amount of time. Once Jenkins starts running several jobs on the same executor, all bets are off.

Reply:
This is where the idea of polling would fit in. That the test would read something like:
```
Given I wait until the values have rotated
```
and the step definition would perform a sleep poll checking for the updated variable versions. If you set your poll interval to the TTL, then you're probably only waiting once or twice, but you are now more resilient to variance in the runtime environment.

With the polling, you probably do want to check that the actual rotate -> rotate duration (From the model timestamps) is actually within an acceptable margin of the TTL.

Reply:
@alan can you elaborate on the Jenkins problem.  I don't believe there is an issue here if you look at the logic.  The test is testing what we actually want it to test -- ie, we actually do need to test that the ttl works.

@micahlee i couldn't tell what you were arguing for here.  but fwiw i'm currently using polling under the hood.  and since i poll every 1/3 of a second, it's guaranteed that i'll see 3 distinct keys if i wait 3 seconds.  if that doesn't happen, we *want* an error reported.